### PR TITLE
Use Accordion component from new macaw

### PR DESCRIPTION
**Pull request by poulch**
closes https://github.com/saleor/saleor-dashboard/issues/3704

Accordion has been changed here:
- Configuration / Channel details right sidebar Warehouses and Shipping Zones
- Products export dialog
- Channel availability card in Collection, Sales, Vouchers, Product, Shipping zone create and details page
- Filter content in all places where filter is used
- Timeline event use for example in order and gift card history
- Product variant details - available channels
 
### Screenshots

| Before | After |
| ------- | ------- |
|  Screenshot 2023-06-28 at 10 10 35 | Screenshot 2023-06-28 at 10 10 48 |

| Before | After |
| ------- | ------- |
|  Screenshot 2023-06-28 at 10 18 11 |  Screenshot 2023-06-28 at 10 18 03 |

| Before | After |
| ------- | ------- |
|  Screenshot 2023-06-28 at 10 23 41 |  Screenshot 2023-06-28 at 10 23 50 |

| Before | After |
| ------- | ------- |
| Screenshot 2023-06-28 at 10 31 27 |  Screenshot 2023-06-28 at 10 31 58 |

| Before | After |
| ------- | ------- |
| Screenshot 2023-06-28 at 10 34 55 |   Screenshot 2023-06-28 at 10 35 30 |

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=6

### CODE REVIEW

File: src/components/Timeline/TimelineEvent.tsx [hunk 1]
```diff
@@ -88,24 +16,50 @@ export const TimelineEvent: React.FC<TimelineEventProps> = props => {
...
+        <Accordion
+          className={sprinkles({
+            width: "100%",
+          })}
+        >
+          <Accordion.Item
+            className={sprinkles({
+              width: "100%",
+            })}
+            value="test"
```
Comment (krzysztofzuraw) [inline src/components/Timeline/TimelineEvent.tsx:47]: **question**: do you need this value here?

File: src/components/Timeline/TimelineEvent.tsx [hunk 2]
```diff
@@ -88,24 +16,50 @@ export const TimelineEvent: React.FC<TimelineEventProps> = props => {
...
-              date={date}
-              titleElements={titleElements}
-              hasPlainDate={hasPlainDate}
-            />
-          </AccordionSummary>
-          <AccordionDetails>
-            <Typography>{children}</Typography>
-          </AccordionDetails>
+        <Accordion
+          className={sprinkles({
```
Comment (krzysztofzuraw) [inline src/components/Timeline/TimelineEvent.tsx:39]: **question**: we don't have `width` defined here? 🤔

**Review by krzysztofzuraw — APPROVED** [review]

**Review by krzysztofzuraw — DISMISSED** [review]

**Review by andrzejewsky — APPROVED** [review]

File: .changeset/beige-dots-protect.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"saleor-dashboard": patch
```
Comment (andrzejewsky) [inline .changeset/beige-dots-protect.md:2]: minor

**Review by andrzejewsky — DISMISSED** [review]

**Review by krzysztofzuraw — DISMISSED** [review]

**Comment by wojteknowacki** [comment]
When selecting filters user sees focus on grid which is underneath 

https://github.com/saleor/saleor-dashboard/assets/124166231/3f48d9b1-f12a-440b-a2bf-e74a5d506713

**Comment by wojteknowacki** [comment]
The whole row could be clickable on filters - now only the txt field and radio button is clickable
image

**Comment by wojteknowacki** [comment]
Closing expanded filter field needs few click actions to work


https://github.com/saleor/saleor-dashboard/assets/124166231/e7048881-130e-4479-8a62-704f0d027a68

**Comment by michalina-graczyk** [comment]
image

**Comment by michalina-graczyk** [comment]
> The whole row could be clickable on filters - now only the txt field and radio button is clickable image

I confirm. This is very unintuitive

**Comment by michalina-graczyk** [comment]
image

**Comment by michalina-graczyk** [comment]
[Untitled_ Jul 4, 2023 2_23 PM.webm](https://github.com/saleor/saleor-dashboard/assets/25908728/ca57d18e-d54d-40c2-b3c2-b4f5710eeda0)
1. Go to orders
2. Go into random order details
3. Go to the order history
4. Click on the arrow - nothing happened

it looks like we implement a dead feature. What should happen when I click there?

**Comment by michalina-graczyk** [comment]
image
What is the first letter in the countries ?

**Comment by michalina-graczyk** [comment]
`Timeline event use for example in order and gift card history` -> can't see it, am I on wrong view?

image
image

**Comment by michalina-graczyk** [comment]
`Product variant details - available channels`
How it should looks on this view? Current:
image

**Comment by michalina-graczyk** [comment]
image

**Review by michalina-graczyk — CHANGES_REQUESTED** [review]
All changes are described in comments

**Comment by poulch** [comment]
> image

This is intentional that accordion content is indented compared to the accordion trigger, and this is how it works currently on staging.

**Comment by poulch** [comment]
> > The whole row could be clickable on filters - now only the txt field and radio button is clickable image
> 
> I confirm. This is very unintuitive

I've decided to not include Filter as it will be replaced be new component

**Comment by poulch** [comment]
> image

As I've described above, Filter is not part of this PR.

**Comment by poulch** [comment]
> Untitled_.Jul.4.2023.2_23.PM.webm 
> 1. Go to orders
> 2. Go into random order details
> 3. Go to the order history
> 4. Click on the arrow - nothing happened
> 
> it looks like we implement a dead feature. What should happen when I click there?

This will be fixed in this PR

**Comment by poulch** [comment]
> image What is the first letter in the countries ?

This is not an accordion component but only looks similar. IMO we can migrate this component during whole page migration to new macaw

**Comment by poulch** [comment]
> `Timeline event use for example in order and gift card history` -> can't see it, am I on wrong view?
> 
> image image

You were in good view but not all history events have accordion.
Example for orders: https://3704-replace-accordion.dashboard.saleor.rocks/orders/T3JkZXI6YzkxZTQzMDMtOGNlOS00ZDRlLTk2OGYtMmZjMGI5NjdmMmIy
I coundn't found an example for gift card but this is the same component as in order history.

**Comment by poulch** [comment]
> `Product variant details - available channels` How it should looks on this view? Current: image

By ``Product variant details ` I mean this view: https://3704-replace-accordion.dashboard.saleor.rocks/products/UHJvZHVjdDo3Nw%3D%3D/variant/UHJvZHVjdFZhcmlhbnQ6MjE3

**Comment by poulch** [comment]
> image

Please create separate issue for this

**Review by michalina-graczyk — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**